src/components/beat.js

```diff
@@ -421,13 +434,25 @@ AFRAME.registerComponent('beat', {
       if (this.beatSystem.data.gameMode === 'punch') {
         bbox.setFromObject(weaponEl.getObject3D('mesh')).expandByScalar(0.1);
         weaponBbox = bbox;
+
+        if (!weaponBbox) { continue; }
+        if (!weaponBbox.intersectsBox(beatBbox)) { continue; }
       } else {
         const blade = weaponEl.components.blade;
-        weaponBbox = blade.boundingBox;
-      }
 
-      if (!weaponBbox) { continue; }
-      if (!weaponBbox.intersectsBox(beatBbox)) { continue; }
+        for (let i = 0; i < 4; i++) {
+          this.bladeLocalPositions [i].copy(blade.bladeWorldPositions [i]);
+          this.blockEl.object3D.worldToLocal(this.bladeLocalPositions [i]);
+        }
+
+        this.bladeLocalTriangles [0].set(this.bladeLocalPositions [0], this.bladeLocalPositions [1], this.bladeLocalPositions [2]);
```

Comment:
i think we can calculate this in blade once per tick so we don't have to do it for each beat.

Reply:
I see, it's done locally in the beat. hmm wonder if it can be optimized

Reply:
how come the triangles need to be projected into the local space if the bbox is in world space?

Reply:
Okay, I'll do some test with that.

I assumed that doing the intersection check within the local coordinate space of the beat would lead to a more precise bounding box since it is not rotated in relation to the controller rig. But I'll try it and adapt the code if it is close enough :)

Reply:
i see, i never tried effectively rotating an AA bounding box that way. just need to wrap my head around it, haha

Reply:
raycasts are also an option, but it might be slower

Reply:
> how come the triangles need to be projected into the local space if the bbox is in world space?

Please bear in mind that I am a bloody beginner in VR / 3D / AFrame, so some (all?) of my obbservations might be wrong. Take it with a grain of salt :)

I inspected the bounding boxes of the meshes and found out they are very tighly constructed around the mesh's vertices. As soon as I get the bounding box using `expandFromObject` it seemed to not be as tighly packed any more since it had to respect rotation along the scene graph. So I decided to translate the blade's tip and base points relative to the local coordinate space of the beat to have the most tightly packed BB to intersect the triangles with.

And since the blades travel a bit between frames it made sense the construct two triangles to create an area between this and last frame's blade tip and base points. The downside is - as you correctly stated - that the blade points have to be projected onto every beat every frame...

I just know tried to do the calculation relative to the `curveFollowRig` but then the beat's bounding boxes again vary in size. I'll continue testing later today and get to you if I find something :)

And I fear I am not confident enough to try a ray casting method since I still would need to cast multiple rays to respect the blade's travel between frames...

Reply:
Code looks really sound and great! Idiomatic three.js / A-Frame.

That all makes sense. I'll test it out more (perf) and will merge if it looks good.